### DISCUSSION TITLE
How about letting people fill out a form via a chatbot like a custom GPT?

### DISCUSSION THREAD

**Discussion by barakiva** [wiserdor/MilCalc#4] [Ideas]
I think converting the form fields into questions asked by a chatbot would make the process easier for some people. Also, it makes the process more accessible for spouses filling the form from a laptop at home (I assume the form is not desktop responsive because most reservists will use a smartphone).

I think the best solution would be either using a custom GPT agent (store opened just a few days ago) that calls to some rudimentary "benefits" API, or a botpress workflow connected to a chat bubble on the app.

**Comment by wiserdor** [comment]
This is a mind blowing idea!
I think it can be great for q&a and questions about the calculations, but for calculating by itself it can be not reliable.

Second concern is that not everyone have access to chatgpt.

The beat of 2 worlds will be to use the api in our site but it's expensive and we don't have server.

Please tell me if you have ideas about how to implement it.

Thank you we appreciate your help 🙂

**Reply by barakiva** [reply]
Hey thank you i'm happy to help!

If you can tell me the daily usage stats, I could email local Israeli startups (or even my company) if they would be willing to act as sponsor for what I assume to be negligible API costs. It's very common for open source projects to place a company logo for as little as 25$ a month. 

With regards to reliability, I suggest adding an Action (under "Code Interpreter" where you edit the GPT)  where the GPT sends an API call for an api that does calculations. That way you get the casual way of collecting information without a form, without giving up reliability.

With regards to not having an API - how about I build it? Will get me to do something since you've already made the GPT 🥲

**Reply by wiserdor** [reply]
i think this one is the most suitable? https://help.openai.com/en/articles/8550641-assistants-api#h_061c53c67a
quite expansive though
do we need to have server for such thing?
do we need to hide our api or can it be public?

**Reply by wiserdor** [reply]
@barakiva please comment here so i can assign you to this task
https://github.com/wiserdor/MilCalc/issues/7

**Reply by barakiva** [reply]
What does the deployment look like? If the VM is big enough, I guess it could run an API together with the frontend app. 

But yes we'd need to have a server. The API won't be big though. Vercel has free hosting so I can try host the API on it.

**Comment by wiserdor** [comment]
added

**Reply by ekurer** [reply]
@barakiva really cool idea and @wiserdor amazing that you added it so quickly. I'm sure you can easily find someone who would be willing to fund to cost to the api calls (maybe on twitter/linkedin) 

![Screenshot 2024-01-14 at 9 13 17](https://github.com/wiserdor/MilCalc/assets/60178602/10e9287d-24df-46e7-b50c-51f046205574)

**Reply by barakiva** [reply]
@ekurer haha yeah i'm lowkey jealous cause I wanted to build it 😞
